libyul/backends/wasm/BinaryTransform.cpp

```diff
@@ -381,30 +376,21 @@ bytes BinaryTransform::operator()(BuiltinCall const& _call)
 		return toBytes(Opcode::I64Const) + lebEncodeSigned(static_cast<int64_t>(m_subModulePosAndSize.at(name).second));
 	}
 
+	yulAssert(builtins.count(_call.functionName), "Builtin " + _call.functionName + " not found");
+	// NOTE: the dialect ensures we have the right amount of arguments
 	bytes args = visit(_call.arguments);
+	bytes ret = move(args) + toBytes(builtins.at(_call.functionName));
+	if (
+		_call.functionName.find(".load") != string::npos ||
+		_call.functionName.find(".store") != string::npos
+	)
+		// Alignment hint and offset. Interpreters ignore the alignment. JITs/AOTs can take it
+		// into account to generate more efficient code but if the hint is invalid it could
+		// actually be more expensive. It's best to hint at 1-byte alignment if we don't plan
+		// to control the memory layout accordingly.
+		ret += bytes{{0, 0}}; // 2^0 == 1-byte alignment
 
-	if (_call.functionName == "unreachable")
-		return toBytes(Opcode::Unreachable);
-	else if (_call.functionName == "nop")
-		return toBytes(Opcode::Nop);
-	else if (_call.functionName == "i32.drop" || _call.functionName == "i64.drop")
-		return toBytes(Opcode::Drop);
```

Comment:
I think having drop here is actually a bug. The arguments were visited, but their results were not stored in the bytecode. This meant we were removing bytecode which could have side-effects.